bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/types/ArithmeticGroupFunctionTest.java

```diff
@@ -201,6 +203,44 @@ public void testSumFunction() {
         assertEquals(new DecimalType("234.95"), state);
     }
 
+    @Test
+    public void testAvgFunction_dimmers() {
+        DimmerItem dimmer1 = new DimmerItem("TestDimmer1");
+        dimmer1.setState(new PercentType(100));
+        DimmerItem dimmer2 = new DimmerItem("TestDimmer2");
+        dimmer2.setState(new PercentType(0));
+
+        items.add(dimmer1);
+        items.add(dimmer2);
+
+        function = new ArithmeticGroupFunction.Avg();
+        State decimalState = function.getStateAs(items, DecimalType.class);
+
+        assertEquals(new DecimalType("50"), decimalState);
```

Comment:
Yes, i was not 100% comfortable with this approach, so I was hoping for some feedback. If you read #1476 , we have an issue where applying a AVG function converts the state value from a PercentType to a Decimal type on dimmers, which is not very useful.   Looking at this again this morning I had another thought.  I could leave the Arithmetic functions alone and instead do a check in GroupItem to convert a DecimalType to PercentType if the base item supports it.  Thoughts?